components/site/tests/site.rs

```diff
@@ -767,6 +770,91 @@ fn can_get_hash_for_static_files() {
+    let json = read_json(&public.join("images/index.html"));
+    let content: [EnqueueResponse; 6] = json::from_value(json).unwrap();
+
+    assert_eq!(content[0], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/3cf8d09c5dc5a58e00.png".into(),
+        static_path: "static/processed_images/3cf8d09c5dc5a58e00.png".into(),
+        width: 150,
+        height: 150,
+        orig_width: 300,
+        orig_height: 380,
+    });
+    assert_eq!(content[1], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/f5150a64b8965f1d00.jpg".into(),
+        static_path: "static/processed_images/f5150a64b8965f1d00.jpg".into(),
+        width: 150,
+        height: 190,
+        orig_width: 300,
+        orig_height: 380
+    });
+    assert_eq!(content[2], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/f5150a64b8965f1d00.jpg".into(),
+        static_path: "static/processed_images/f5150a64b8965f1d00.jpg".into(),
+        width: 150,
+        height: 190,
+        orig_width: 300,
+        orig_height: 380
+    });
+    assert_eq!(content[3], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/f5150a64b8965f1d00.jpg".into(),
+        static_path: "static/processed_images/f5150a64b8965f1d00.jpg".into(),
+        width: 150,
+        height: 190,
+        orig_width: 300,
+        orig_height: 380
+    });
+    assert_eq!(content[4], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/f5150a64b8965f1d00.jpg".into(),
+        static_path: "static/processed_images/f5150a64b8965f1d00.jpg".into(),
+        width: 150,
+        height: 190,
+        orig_width: 300,
+        orig_height: 380
+    });
+    assert_eq!(content[5], EnqueueResponse {
+        url: "https://replace-this-with-your-url.com/processed_images/82443417419c988d00.jpg".into(),
+        static_path: "static/processed_images/82443417419c988d00.jpg".into(),
+        width: 100,
+        height: 200,
+        orig_width: 300,
+        orig_height: 380
+    });
+}
+
+#[test]
+fn can_get_image_metadata() {
+    // Unfortunately we have to use a response with an owned `String` rather than a `&'static str`,
+    // as that can't be deserialized.
+    #[derive(Debug, Deserialize, PartialEq, Eq)]
+    struct ImageMetaResponse {
+        width: u32,
+        height: u32,
+        format: Option<String>,
+    }
+
+    let (_, _tmp_dir, public) = build_site("test_site");
+    let json = read_json(&public.join("images/index.html"));
+    let content: [ImageMetaResponse; 2] = json::from_value(json).unwrap();
+
+    assert_eq!(content[0], ImageMetaResponse {
+        width: 300,
+        height: 380,
+        format: Some("png".into()),
+    });
+    assert_eq!(content[1], ImageMetaResponse {
+        width: 300,
+        height: 380,
+        format: Some("jpg".into()),
+    });
+}
+
```

Comment:
It all looks good to me except that - can we somehow move the tests in the imageproc module? This component already so many random tests that it's a bit hard to follow

Reply:
Sure! Will do :slightly_smiling_face:

Reply:
@Keats Moved the `imageproc` tests into its crate.
(But tbh it's kind of late over here and I should probably double-check everything tomorrow :slightly_smiling_face:)